src/components/RadioGroup/RadioGroup.stories.tsx

```diff
@@ -36,3 +46,56 @@ export const Basic: React.FC = () => {
     </form>
   );
 };
+
+export const WithRules: React.FC = () => {
+  const { control, handleSubmit, reset, errors, watch } = useForm<ITestProps>();
+  const currentValue = watch('campaign');
+
+  return (
+    <form
+      onSubmit={handleSubmit((payload) => {
+        window.alert(JSON.stringify(payload));
+      })}
+    >
+      <Text as="p">Current Value: {currentValue}</Text>
+      <Separator orientation="horizontal" verticalSpacing="space30" />
+
+      {errors.campaign?.message && <Alert variant="error">{errors.campaign.message}</Alert>}
+
+      <RadioGroup<ITestProps>
+        id="campaign"
+        name="campaign"
+        legend="When should your campaign run?"
+        control={control}
+        controllerProps={{
+          defaultValue: '',
+          rules: {
+            validate: (value) => {
+              if (!value.length) {
+                return 'Please provide a value.';
+              }
```

Comment:
According to the Paste radio group guidelines, your use-case of adding a required validator isn't natively supported; in that case, to solve this I used a custom `validate` function here to handle that case non-natively.

> HTML radio groups don't natively support the required attribute.

https://paste.twilio.design/components/radio-group#accessibility-information

Reply:
@vnguyen94, this is good but in context of `react-hook-form` the rules already reports to the formState. Please consider this code we use for our form.

`
<RadioGroup
            id="method-group"
            name="Method"
            legend={webhookLabels.methodTitle}
            control={control}
            rules={{ required: true }}
            defaultValue=""
            required
          >
            <Radio
              id="method-post"
              data-testid="method-post"
              value="post"
              name="Method"
            >
              {webhookLabels.methodPost}
            </Radio>

            <Radio
              id="method-get"
              data-testid="method-get"
              value="get"
              name="Method"
            >
              {webhookLabels.methodGet}
            </Radio>
          </RadioGroup>
          {errors.Method && (
            <HelpText variant="error" data-testid="method-error-helptext">
              {webhookLabels.methodFieldValidationError}
            </HelpText>
          )}
`

Reply:
to re-iterate, `rules={{ required: true }}` will _not_ work; it is not supported by native HTML validation. you'd have to copy my validate function to fit your needs

Reply:
for reference i tried your code in my storybook and it didn't work

Reply:
It should work if you provide a string instead of boolean in storybook. 

`rules={{ required: 'You must provide a value' }}`

Reply:
alrighty then i revoke everything i said above, it's working on my local now. apologies, looks great!